plugins/eos/gs-plugin-eos.c

```diff
@@ -67,6 +67,56 @@ gs_plugin_adopt_app (GsPlugin *plugin, GsApp *app)
 	gs_app_set_management_plugin (app, plugin);
 }
 
+static void
+refine_core_app (GsApp *app)
+{
+	if (app_is_flatpak (app) ||
+	    (gs_app_get_scope (app) == AS_COMPONENT_SCOPE_UNKNOWN))
+		return;
+
+	if (gs_app_get_kind (app) == AS_COMPONENT_KIND_OPERATING_SYSTEM)
+		return;
+
+	/* Hide non-installed apt packages, as they can’t actually be installed.
+	 * The installed ones are pre-installed in the image, and can’t be
+	 * removed. We only allow flatpaks to be removed. */
+	if (!gs_app_is_installed (app)) {
+		gs_app_add_quirk (app, GS_APP_QUIRK_HIDE_EVERYWHERE);
+	} else {
+		gs_app_add_quirk (app, GS_APP_QUIRK_COMPULSORY);
+	}
```

Comment:
OK, this plugin does end up doing something more interesting than "launch an app".